schema/spacewalk/postgres/procs/insert_product_vulnerable_packages.sql

```diff
@@ -0,0 +1,63 @@
+--
+-- Copyright (c) 2023 SUSE LLC
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+-- Red Hat trademarks are not licensed under GPLv2. No permission is
+-- granted to use or replicate Red Hat trademarks that are incorporated
+-- in this software or its documentation.
+--
+
+CREATE OR REPLACE PROCEDURE
+insert_product_vulnerable_packages(package_name_in varchar,fix_version_in varchar,product_cpe_in varchar,cve_name_in varchar)
+AS
+$$
+DECLARE
+    cve_id_val numeric;
+    product_cpe_id_val numeric;
+    vulnerable_pkg_id_val numeric;
+begin
+
+    IF NOT exists(SELECT cve FROM rhnCve cve WHERE cve.name = cve_name_in) THEN
+        INSERT INTO rhncve(id, name)
+        VALUES (nextval('rhn_cve_id_seq'), cve_name_in);
```

Comment:
why not using `lookup_cve(cve_name_in)` - this is handling also creating of the value if it does not exist.

Reply:
I didn't know about it. But now that I look into it and know that it calls `insert_cve`, it will lead to CVE IDs growing exponentially, because of the calling of `nextval('rhn_cve_id_seq')` even when CVE already exists.

This is an [instance of the problem](https://github.com/uyuni-project/uyuni/pull/7466#discussion_r1366811006) that is now fixed in the suseOVALPlatform table.

![image](https://github.com/uyuni-project/uyuni/assets/21198231/fab8442a-de76-41d0-9c04-4adb720faa90)

Reply:
But insert is only called when a select failed because the entry does not exist

Reply:
Ah okay, thank you for clarifying; I made the change.